src/services/createRouterPlugin.ts

```diff
@@ -0,0 +1,42 @@
+import { Routes } from "@/types/route";
+import { Router } from "@/types/router";
+import { RouterPlugin } from "@/types/routerPlugin";
+import { asArray } from "@/utilities/array";
+import { Component } from "vue";
+
+export function createRouterPlugin<
+  TRoutes extends Routes = [], 
+  TRejections extends Record<string, Component> = {}
+>(plugin: Partial<RouterPlugin<TRoutes, TRejections>>): RouterPlugin<TRoutes, TRejections> {
+  return {
+    routes: plugin.routes ?? [] as unknown as TRoutes,
+    rejections: plugin.rejections ?? {} as TRejections,
+    ...plugin,
+  }
+}
+
+export function addRouterPluginHooks(router: Router, plugin: RouterPlugin) {
```

Comment:
looks like maybe linting is broken in your vscode, there should be a return type on this function

Reply:
I have a open question about why hooks are treated slightly differently in createRouter. If we do in fact need this, I don't think it belongs in this file.

Reply:
Hooks are always added to the hooks store. Global hooks are added by calling the routers hook methods like `router.onBeforeRouteEnter(() => {...})`. So somewhere when we're "installing" a plugin we need to add those hooks to the store. 

Where do you think this sort of logic belongs?

Reply:
We're going to follow up on this once we do some investigation into global hooks.